src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -42,19 +52,47 @@ internal abstract class AbstractSnippetExpansionClient : ForegroundThreadAffinit
         protected readonly ITextView TextView;
         protected readonly ITextBuffer SubjectBuffer;
 
+        private readonly IEnumerable<Lazy<ArgumentProvider, OrderableLanguageMetadata>> _allArgumentProviders;
```

Comment:
nit, consider storing this as an ImmutbaleArray.

Reply:
⏱️ Now using ImmutableArray